backend/audit/models.py

```diff
@@ -50,7 +52,44 @@ def create(self, **obj_data):
         return super().create(**updated)
 
 
-class SingleAuditChecklist(models.Model):
+def camel_to_snake(raw: str) -> str:
+    """Convert camel case to snake_case."""
+    text = f"{raw[0].lower()}{raw[1:]}"
+    return "".join(c if c.islower() else f"_{c.lower()}" for c in text)
+
+
+def json_property_mixin_generator(name, fname=None, toplevel=None, classname=None):
+    """Generates a mixin class named classname, using the top-level fields
+    in the properties field in the file named fname, accessing those fields
+    in the JSON field named toplevel.
+    If the optional arguments aren't provided, generate them from name."""
+    filename = fname or f"{name}.schema.json"
+    toplevelproperty = toplevel or camel_to_snake(name)
+    mixinname = classname or f"{name}Mixin"
+
+    def _wrapper(key):
+        def inner(self):
+            try:
+                return getattr(self, toplevelproperty)[key]
+            except KeyError:
+                pass
```

Comment:
do you think it'd be worth adding some logging in these exception handlers? This looks like a place where problems could be tricky (or at least tedious) to pin down